src/app/TideIndexApplication.cpp

```diff
@@ -1,1183 +1,1207 @@
+	  int systemResult = system(cmd.c_str());
+	  if (systemResult != 0)
+		  carp(CARP_FATAL, "System sort failed, the call returned code: %i", systemResult);
+    #else
+      std::cout << "OS not supported";
+	  carp(CARP_FATAL, "OS not supported");
+	  std::string cmd = ""
+    #endif
+		 
+    bool is_exist = boost::filesystem::exists(pathSortedPeptideFile);
+	  if (is_exist) {
+		  uintmax_t sorted_peptide_file_size = boost::filesystem::file_size(pathSortedPeptideFile);
+		  uintmax_t unsorted_peptide_file_size = boost::filesystem::file_size(pathSortedPeptideFile);
+		  if (sorted_peptide_file_size != unsorted_peptide_file_size) {
+			  carp(CARP_FATAL, "Something went wrong whilst sorting, because sorted peptide file and peptide files have different sizes");
+		  }
+	  }else {
+		  carp(CARP_FATAL, "System sort failed to create the sorted petide file");
+	  }
+  } else {
+    sort(peptide_list.begin(), peptide_list.end(), less<TideIndexPeptide>());    
+  }
+ 
+  // Prepare the protocol buffer for the peptides.  
+  carp(CARP_INFO, "Writting peptides");
+
+  // pb::Header header_with_mods;
+  pb::Header_PeptidesHeader& pep_header = *(header_with_mods.mutable_peptides_header());
+  
+  pep_header.Clear();
+  pep_header.set_min_mass(min_mass);
+  pep_header.set_max_mass(max_mass);
+  pep_header.set_min_length(min_length);
+  pep_header.set_max_length(max_length);
+  pep_header.set_monoisotopic_precursor(monoisotopic_precursor);
+  pep_header.set_enzyme(enzyme);
+  if (enzyme != "no-enzyme") {
+    pep_header.set_full_digestion(digestion == FULL_DIGEST);
+    pep_header.set_max_missed_cleavages(missed_cleavages);
+  }
+  pep_header.mutable_mods()->CopyFrom(*(var_mod_table.ParsedModTable()));
+  pep_header.mutable_nterm_mods()->CopyFrom(*(var_mod_table.ParsedNtpepModTable()));
+  pep_header.mutable_cterm_mods()->CopyFrom(*(var_mod_table.ParsedCtpepModTable()));
+  pep_header.mutable_nprotterm_mods()->CopyFrom(*(var_mod_table.ParsedNtproModTable()));
+  pep_header.mutable_cprotterm_mods()->CopyFrom(*(var_mod_table.ParsedCtproModTable()));
+
+  pep_header.set_decoys_per_target(numDecoys);
+
+  header_with_mods.set_file_type(pb::Header::PEPTIDES);
+  header_with_mods.set_command_line(cmd_line);
+  pb::Header_Source* source = header_with_mods.add_source();
+  source->mutable_header()->CopyFrom(proteinPbHeader);
+  source->set_filename(AbsPath(out_proteins));
+
+  pb::Header header_no_mods;
+  header_no_mods.CopyFrom(header_with_mods);
+  pb::ModTable* del = header_no_mods.mutable_peptides_header()->mutable_mods();
+  del->mutable_variable_mod()->Clear();
+  del->mutable_unique_deltas()->Clear();
+
+  bool need_mods = var_mod_table.Unique_delta_size() > 0;
+
+  string peptidePbFile = need_mods ? modless_peptides : peakless_peptides;  
+  
+  // Check header
+  if (header_no_mods.source_size() != 1) {
+    carp(CARP_FATAL, "header_no_mods had a number of sources other than 1");

```

Comment:
In general, user-level messages should not refer to elements of the code (header_no_mods). Also, I don't understand what this means.